### PR TITLE
Fix Doctrine subscriber deprecations

### LINKED ISSUE #736: Symfony 6.3 deprecation doctrine subscribers
**Sylius version affected**: > 1.10

**Description**  

User Deprecated: Since symfony/doctrine-bridge 6.3: Using Doctrine subscribers as services is deprecated, declare listeners instead

Deprecation message is triggered for the following classes:
- Sylius\Bundle\ResourceBundle\EventListener\ORMTranslatableListener
- Sylius\Bundle\ResourceBundle\EventListener\ORMRepositoryClassSubscriber
- Sylius\Bundle\ResourceBundle\EventListener\ORMMappedSuperClassSubscriber

### PR DESCRIPTION
**Pull request by loic425**
| Q               | A
| --------------- | -----
| Bug fix?        | yes (deprecations)
| New feature?    | no
| BC breaks?      | no
| Deprecations?   | yes and UPGRADE.md has been updated
| Related tickets | fixes #736
| License         | MIT

### CODE REVIEW

**Review by loevgaard — APPROVED** [review]

**Comment by loic425** [comment]
@lchrusciel Maybe should we keep the public methods for user decorations.

File: src/Bundle/EventListener/ORMRepositoryClassSubscriber.php [hunk 1]
```diff
@@ -15,18 +15,10 @@
 
 use Doctrine\ORM\EntityRepository;
 use Doctrine\ORM\Event\LoadClassMetadataEventArgs;
-use Doctrine\ORM\Events;
 use Doctrine\ORM\Mapping\ClassMetadata;
 
-final class ORMRepositoryClassSubscriber extends AbstractDoctrineSubscriber
+final class ORMRepositoryClassSubscriber extends AbstractDoctrineListener
 {
-    public function getSubscribedEvents(): array
```
Comment (lchrusciel) [inline src/Bundle/EventListener/ORMRepositoryClassSubscriber.php:23]: Public methods must stay and become deprecated

File: src/Bundle/EventListener/ORMTranslatableListener.php [hunk 2]
```diff
@@ -26,7 +24,7 @@
 use Sylius\Component\Resource\Translation\TranslatableEntityLocaleAssignerInterface;
 use Symfony\Component\DependencyInjection\ContainerInterface;
 
-final class ORMTranslatableListener implements EventSubscriber
+final class ORMTranslatableListener
```
Comment (lchrusciel) [inline src/Bundle/EventListener/ORMTranslatableListener.php]:
```suggestion
final class ORMTranslatableListener implements AbstractDoctrineSubscriber
```

Comment (loic425) [reply src/Bundle/EventListener/ORMTranslatableListener.php:27]: Do you meen extends AbstractDoctrineListener?

**Review by lchrusciel — APPROVED** [review]

**Comment by lchrusciel** [comment]
Thank you, @loic425!

**Pull request merged**